langwatch/src/pages/api/cron/scenario_analytics.ts

```diff
@@ -0,0 +1,224 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import { prisma } from "~/server/db";
+import { esClient } from "~/server/elasticsearch";
+import { type Prisma } from "@prisma/client";
+import { ANALYTICS_KEYS } from "~/types";
+import { createScenarioAnalyticsQueriesForAllEventTypes } from "~/server/scenario-analytics";
+
+export default async function handler(
```

Comment:
There's a lot going on in this function. Consider breaking things into smaller functions so it's easier to read/test.